### PR TITLE
Fix CF template if asgMetricsCollection is having no metrics

### PR DESCRIPTION
**Pull request by sayboras**
### Description

Fixes https://github.com/weaveworks/eksctl/issues/2265

### Checklist
- [x] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [ ] Added labels for change area (e.g. `area/nodegroup`), target version (e.g. `version/0.12.0`) and kind (e.g. `kind/improvement`)
- [x] Make sure the title of the PR is a good description that can go into the release notes

### CODE REVIEW

File: pkg/cfn/builder/nodegroup.go [hunk 1]
```diff
@@ -370,9 +370,9 @@ func metricsCollectionResource(asgMetricsCollection []api.MetricsCollection) []m
 	for _, m := range asgMetricsCollection {
 		newCollection := make(map[string]interface{})
 
-		var metrics []string
-		metrics = append(metrics, m.Metrics...)
-		newCollection["Metrics"] = metrics
+		if len(m.Metrics) > 0 {
+			newCollection["Metrics"] = m.Metrics
```
Comment (michaelbeaumont) [inline pkg/cfn/builder/nodegroup.go:374]: The bug is that we have `newCollection["Metrics"] = nil`, right? `[]` seems to be a valid value for CFN.

I think comparing to `nil` should be enough?

Comment (sayboras) [reply pkg/cfn/builder/nodegroup.go:374]: Thanks for your input. I am not sure about the effect of passing [] in CF, will it fail for validation, or not ? So to be safe, I just proceed to with _len()_ (which internally check for nil slice as well).

If you have a chance to confirm the above, I can make a chance. Result will be the same from end user 's point of view.

Comment (michaelbeaumont) [reply pkg/cfn/builder/nodegroup.go:374]: I guess the question is what does the user expect in the case where they pass `[]` explicitly, no metrics or all metrics?

Comment (sayboras) [reply pkg/cfn/builder/nodegroup.go:374]: I can't test it out right now as I have no credit in my aws account. My intepretation on aws docs for this case is _all metrics_

Comment (michaelbeaumont) [reply pkg/cfn/builder/nodegroup.go:374]: indeed, seems to be the case that `null` and `[]` are treated the same by cfn.

**Pull request merged**